tests/e2e_application_aware_resource_quota_test.go

```diff
@@ -1364,107 +1365,143 @@ var _ = Describe("ApplicationAwareResourceQuota", func() {
-			return err
-		}, 2*time.Minute, 1*time.Second).Should(BeNil())
-		utils.VerifyPodIsNotGated(f.K8sClient, f.Namespace.Name, pod.Name)
+			}).WithTimeout(30*time.Second).WithPolling(5*time.Second).ShouldNot(HaveOccurred(), "cannot remove the NoSchedule taint from the node")
+		})
 
-		By("Ensuring that the pod has proper node affinity")
-		Eventually(func() error {
-			pod, err = f.K8sClient.CoreV1().Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
-			if err != nil {
+		It("should skip .spec.nodeName", Serial, func(ctx context.Context) {
+			By("Counting existing ApplicationAwareResourceQuota")
+			c, err := countApplicationAwareResourceQuota(ctx, f.AaqClient, f.Namespace.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a ApplicationAwareResourceQuota")
+			quotaName := "test-quota"
+			ApplicationAwareResourceQuota := newTestApplicationAwareResourceQuota(quotaName)
+			ApplicationAwareResourceQuota, err = createApplicationAwareResourceQuota(ctx, f.AaqClient, f.Namespace.Name, ApplicationAwareResourceQuota)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring Application Aware Resource Quota status is calculated")
+			usedResources := v1.ResourceList{}
+			usedResources[v1.ResourceQuotas] = resource.MustParse(strconv.Itoa(c + 1))
+			err = waitForApplicationAwareResourceQuota(ctx, f.AaqClient, f.Namespace.Name, quotaName, usedResources)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a Pod that doesn't fits quota")
+			podName := "test-pod"
+			requests := v1.ResourceList{}
+			requests[v1.ResourceMemory] = resource.MustParse("600Mi") //quota has only 500Mi
+			pod := utils.NewTestPodForQuota(podName, requests, nil)
+			pod.Spec.NodeSelector = nil
+			pod.Spec.NodeName = *nodeName
+			pod, err = f.K8sClient.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
+
+			Expect(err).ToNot(HaveOccurred())
+			utils.VerifyPodIsGated(f.K8sClient, f.Namespace.Name, pod.Name)
+
+			By("Update arq to fit pod")
+			Eventually(func() error {
+				currApplicationAwareResourceQuota, err := f.AaqClient.AaqV1alpha1().ApplicationAwareResourceQuotas(f.Namespace.Name).Get(ctx, ApplicationAwareResourceQuota.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				currApplicationAwareResourceQuota.Spec.Hard[v1.ResourceMemory] = resource.MustParse("600Mi")
+				_, err = f.AaqClient.AaqV1alpha1().ApplicationAwareResourceQuotas(f.Namespace.Name).Update(ctx, currApplicationAwareResourceQuota, metav1.UpdateOptions{})
 				return err
-			}
+			}, 2*time.Minute, 1*time.Second).Should(BeNil())
+			utils.VerifyPodIsNotGated(f.K8sClient, f.Namespace.Name, pod.Name)
+
+			By("Ensuring that the pod has proper node affinity")
+			Eventually(func() error {
+				pod, err = f.K8sClient.CoreV1().Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
```

Comment:
we can close the Eventually scope after this if statement

Reply:
What if the affinity is not updated yet? I think we want to test it for a while to ensure there's no race in the tests that will make them flaky